sysutils/pfSense-pkg-LCDproc/files/usr/local/pkg/lcdproc_client.php

```diff
@@ -1339,6 +1390,52 @@ function loop_status($lcd) {
 					$lcd_cmds[] = "widget_set {$name} title_wdgt 1 1 \"{$title}\"";
 					$lcd_cmds[] = "widget_set {$name} text_wdgt 1 2 {$lcdpanel_width} 2 h 4 \"{$cputemperature}\"";
 					break;
+				case "scr_ntp":
+					$ntp_hms = date("H:i");
+					$ntpq_output = lcdproc_get_ntp_status();
+					if ($ntpq_output != false) {
+						$ntptime = 
+						$ntpq_counter = 0;
+						foreach ($ntpq_output as $line) {
+							if (substr($line, 0, 1) == "*" || substr($line, 0, 1) == "o") {
+								//Active NTP Peer
+								$line = substr($line, 1);
+								$peerinfo = preg_split("/[\s\t]+/", $line);
+								$peerdel = substr($peerinfo[7],0,5);
+								$peeroff = substr($peerinfo[8],0,6);
+								$peerjit = substr($peerinfo[9],0,5);
+								if ($peerinfo[2] == "0" && $peerinfo[1] == ".GPS.") {
+									// When local GPS is used. Show the IP of the source and the stratum. Also append how many satellites are in use or in view
+									$gps_sat_count = lcdproc_get_ntp_gps_sat_count();
+									$lcd_cmds[] = "widget_set {$name} time_st_wdgt 1 1 {$lcdpanel_width} 2 h 4 \"NTP: {$ntp_hms} Strtm: {$peerinfo[2]}\"";
+									$lcd_cmds[] = "widget_set {$name} ref_wdgt 1 2 {$lcdpanel_width} 2 h 4 \"Ref: {$peerinfo[1]} " . (($gps_sat_count == false) ? "" : "{$gps_sat_count}") . "\"";
+									$lcd_cmds[] = "widget_set {$name} text_wdgt 1 3 {$lcdpanel_width} 2 h 4 \"Delay Offset Jitter\"";
+									$lcd_cmds[] = "widget_set {$name} stats_wdgt 1 4 {$lcdpanel_width} 2 h 4 \"{$peerdel} {$peeroff} {$peerjit}\"";
+								} else {
+									// Show IP of the source and the stratum
+									$lcd_cmds[] = "widget_set {$name} time_st_wdgt 1 1 {$lcdpanel_width} 2 h 4 \"NTP: {$ntp_hms} Strtm: {$peerinfo[2]}\"";
+									$lcd_cmds[] = "widget_set {$name} ref_wdgt 1 2 {$lcdpanel_width} 2 h 4 \"Ref: {$peerinfo[1]}\"";
```

Comment:
There is a similar question here as above but this one does seem like it should be `$peerinfo[0]` so it prints the IP address. Right now when testing it's showing me things like `.PPS.` and `.XMIS.` and not the IP address of the active peer.

Reply:
I think you are right.  Now that I think about it, my recollection is the server column is the actual IP address of the polled NTP server, but the reference column is the source the polled NTP server is actually using as its reference.   I think I would have add another elseif test for a Stratum 0 .PPS. source to display $peerinfo[1], then the switch the last else to use $peerinfo[0] for everything else.  I was also considering unpacking the rest of the peerinfo[n] variables to make it more readable.

Reply:
You are correct, the "ref" column that has `.GPS.` will show the reference clock from which a source derives its time. For servers that are on higher strata they would show the IP address of whichever server they are peered with for time counting. So it could have an IP address there but it's not the one most users would want to see in that case.

Reply:
Submitted updated lcdproc_client.php